src/Form/Type/Filter/ChoiceType.php

```diff
@@ -35,6 +35,12 @@ class ChoiceType extends AbstractType
 
     public const TYPE_EQUAL = 3;
 
+    public const CHOICES = [
+        'label_type_contains' => self::TYPE_CONTAINS,
```

Comment:
IMO, creating a collection of possible values is fine, but the labelling responsibility should be delegated to something related with the presentation.

Reply:
@phansys I agree that's not the best but I didn't see better way.
Maybe the way I create filter are not the good way.

Actually I do like this
```
->add('foo', 'doctrine_orm_callback', [
                    'callback'         => [$this, 'myCallBack'],
                    'operator_type'    => Symfony/ChoiceType::class,
                    'operator_options' => [
                        'choices'            => [
                            'label_type_equal'        => Sonata/ChoiceType::TYPE_EQUAL,
                            'label_type_contains'     => Sonata/ChoiceType::TYPE_CONTAINS,
                            'label_type_not_contains' => Sonata/ChoiceType::TYPE_NOT_CONTAINS,
                        ],
                        'translation_domain' => 'SonataAdminBundle',
                    ],
                ],
                ChoiceType::class,
                [
                    'choices'                   => $this->getUser()->getAllFooNames(),
                    'choice_translation_domain' => false,
                    'multiple'                  => true,
                ]
            )
```

I have copy pasted so much time the array
```
'choices'            => [
                            'label_type_equal'        => Sonata/ChoiceType::TYPE_EQUAL,
                            'label_type_contains'     => Sonata/ChoiceType::TYPE_CONTAINS,
                            'label_type_not_contains' => Sonata/ChoiceType::TYPE_NOT_CONTAINS,
                        ],
```

I was looking for a better way.

Reply:
This is the same for the NumberType, I can't use it like this
```
->add('foo', 'doctrine_orm_callback', [
                    'callback'         => [$this, 'myCallBack'],
                ],
                NumberType::class
            )
```

Because it puts me two field, type and value.

In my query I can now use `NumberFilter::CHOICES[$value['field']] ?? '='` but for the declaration of the filter I didn't find better way than

```
->add('foo', 'doctrine_orm_callback', [
                    'callback'         => [$this, 'myCallBack'],
                    'operator_type'    => Symfony/ChoiceType::class,
                    'operator_options' => [
                        'choices'            => [
                            'label_type_equal'        => Sonata/NumberType::TYPE_EQUAL,
                            ...
                        ],
                        'translation_domain' => 'SonataAdminBundle',
                    ],
                ],
            )
```

Reply:
@phansys Is this PR ok for you or do you have another/better solution ? :)

Reply:
You could use this 
https://symfony.com/doc/current/reference/forms/types/choice.html#choice-label

Reply:
@core23 So, if I use ten times the ChoiceType, I have to set ten times the choice label ?

When I use the Sonata EqualType,
```
->add(
                'country',
                CallbackFilter::class,
                [
                    'callback'      => ...,
                    'operator_type' => EqualType::class,
                ],
```
The labelling responsibility is delegated to Sonata

So when I use the ChoiceType
```
->add(
                'contractTypeCombination',
                CallbackFilter::class,
                [
                    'callback'         => ...,
                    'operator_type'    => ChoiceType::class,
                    'operator_options' => [
                        'choices' => ChoiceFilterType::CHOICES,
                        'translation_domain' => 'SonataAdminBundle',
                    ],
                ],
```
Seems to be the same delegation level for me; and avoid duplicated code.

Maybe it's better if I create a const with both CHOICES and translation domain, used like this
```
->add(
                'contractTypeCombination',
                CallbackFilter::class,
                [
                    'callback'         => ...,
                    'operator_type'    => ChoiceType::class,
                    'operator_options' => ChoiceFilterType::OPERATOR_OPTIONS,
                ],
```